src/app/migration/src/src/app/modules/announcement/components/create/create.component.ts

```diff
@@ -0,0 +1,481 @@
+   *
+   * @param {GeoLocationDetails} item selected location details
+   */
+  removeRecipient(item: GeoLocationDetails): void {
+    _.remove(this.recipientsList, (recipient) => {
+      if (recipient.id === item.id) {
+        item.selected = false;
+        this.iziToast.info(item.location + ' ' + this.resource.messages.imsg.m0020);
+        return recipient.location;
+      }
+    });
+
+    if (this.recipientsList.length === 0) {
+      this.navigateToWizardNumber(2);
+    }
+  }
+
+  /**
+   * Executed when user click on preview announcement button
+   *
+   * It helps user to show announcement preview - after creating the announcement how it will look
+   */
+  navigateToPreviewPage(): void {
+    const data = { ...this.announcementForm.value };
+    data.links = data.links.length ? _.map(data.links, 'url') : [];
+    this.announcementDetails = data;
+    const atta = { ...this.attachments };
+    this.announcementDetails.attachments = this.fileUpload.attachedFiles;
+    this.navigateToWizardNumber(4);
+  }
+
+  /**
+   * To enable select recipients button.
+   *
+   * Button gets enabled when all required fields and
+   * at least one of 'Description' or 'URL / weblink' or 'Attachment' must be provided
+   */
+  enableRecipientsBtn(): boolean {
+    const data = this.announcementForm ? this.announcementForm.value : '';
+    if (this.announcementForm.status === 'VALID') {
+      if (data.links.length || data.description || this.fileUpload.attachedFiles && this.fileUpload.attachedFiles.length > 0) {
+        return this.formErrorFlag = false;
+      } else {
+        return this.formErrorFlag = true;
+      }
+    } else {
+      return this.formErrorFlag = true;
+    }
+  }
+
+  /**
+   * It gets executed when user directly hit the url.
+   *
+   * It navigates user to the first step if required fields data are not filled
+   */
+  validateFormState(): void {
+    const data = this.announcementForm.value;
+    if (this.announcementForm.status === 'VALID') {
+      if (data.links.length || data.description || this.fileUpload.attachedFiles && this.fileUpload.attachedFiles.length > 0) {
+      } else {
+        this.navigateToWizardNumber(1);
+      }
+    } else {
+      this.navigateToWizardNumber(1);
+    }
+  }
+
+  /**
+   * Callback function gets called from FileUploadService when user click on remove icon
+   *
+   * Used to remove uploaded file details from local variable
+   */
+  onFileUploadCancel = (id: number, name: string): void => {
+    const data = this.attachments.splice(id, 1);
+    if (data.length === 0) {
+      _.forEach(this.attachments, (value, key) => {
+        if (value.name === name) {
+          this.attachments.splice(key, 1);
+        }
+      });
+    }
+    this.enableRecipientsBtn();
+  }
+
+  /**
+   * Post announcement form data
+   */
+  saveAnnouncement() {
+    const data = { ...this.announcementForm.value };
+    data.links = data.links.length ? _.map(data.links, 'url') : [];
```

Comment:
do we need all these steps, cant we use this.announcementDetails which will be set in step 3.